### PR TITLE
dev-lang/python: add PGO

### PR DESCRIPTION
**Pull request by thesamesam**
TODO: use the full PGO target

Bug: https://bugs.gentoo.org/615412
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: dev-lang/python/python-3.10.0_p1.ebuild [hunk 1]
```diff
@@ -178,6 +183,9 @@ src_configure() {
 		--without-ensurepip
 		--with-system-expat
 		--with-system-ffi
+
+		$(use_with lto)
```
Comment (mgorny) [inline dev-lang/python/python-3.10.0_p1.ebuild:187]: Wrong commit.

File: dev-lang/python/python-3.11.0_alpha2.ebuild [hunk 2]
```diff
@@ -198,6 +206,18 @@ src_compile() {
 	# https://bugs.gentoo.org/823728
 	export SETUPTOOLS_USE_DISTUTILS=stdlib
 
+	if use pgo ; then
+		# bug 660358
+		local -x COLUMNS=80
+		local -x PYTHONDONTWRITEBYTECODE=
+
+		local jobs=$(makeopts_jobs "${MAKEOPTS}" "$(get_nproc)")
+		export PROFILE_TASK="-m test -j${jobs} --pgo-extended -x test_gdb"
```
Comment (mgorny) [inline dev-lang/python/python-3.11.0_alpha2.ebuild:215]: You seem to be doing this twice in this ebuild.

**Review by mgorny — CHANGES_REQUESTED** [review]
Could you please limit the changes to just one ebuild for the purpose of review? I'm getting confused which version is the correct one.

Also, how about changing to `addpredict` first?

File: dev-lang/python/python-3.10.0_p1.ebuild [hunk 3]
```diff
@@ -225,7 +225,7 @@ src_test() {
 	local -x COLUMNS=80
 	local -x PYTHONDONTWRITEBYTECODE=
 	# workaround https://bugs.gentoo.org/775416
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply dev-lang/python/python-3.10.0_p1.ebuild:227]: You're right on the comment, but sadly, we really do need addpredict still.

**Review by mgorny — APPROVED** [review]
LGTM. I don't care about pgo/lto much but please backport the `addpredict` change to all versions.

**Pull request closed without merging**